Model/Api/UpdateCartCommon.php

```diff
@@ -0,0 +1,284 @@
+                if ($this->orderHelper->getExistingOrder($incrementId)) {
+                    $this->sendErrorResponse(
+                        BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                        sprintf('The order #%s has already been created.', $incrementId),
+                        422
+                    );
+                    return false;
+                }
+    
+                // check the existence of child quote
+                $immutableQuote = $this->cartHelper->getQuoteById($immutableQuoteId);
+                if (!$immutableQuote) {
+                    $this->sendErrorResponse(
+                        BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                        sprintf('The cart reference [%s] is not found.', $immutableQuoteId),
+                        404
+                    );
+                    return false;
+                }
+    
+                // check if cart is empty
+                if (!$immutableQuote->getItemsCount()) {
+                    $this->sendErrorResponse(
+                        BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                        sprintf('The cart for order reference [%s] is empty.', $immutableQuoteId),
+                        422
+                    );
+    
+                    return false;
+                }
+                
+                return [
+                    $parentQuote,
+                    $immutableQuote,
+                ];
+
+            } catch (\Exception $e) {
+                $this->bugsnag->notifyException($e);
+                $this->sendErrorResponse(
+                    BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                    sprintf('The cart reference [%s] is not found.', $parentQuoteId),
+                    404
+                );
+                return false;
+            }
+        } else {
+            $this->bugsnag->notifyError(
+                BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                'The cart.order_reference is not set or empty.'
+            );
+            $this->sendErrorResponse(
+                BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                'The cart reference is not found.',
+                404
+            );
+            return false;
+        }
+    }
+    
+    /**
+     *
+     * Set the shipment if request payload has that info.
+     * 
+     * @param array $shipment
+     * @param Quote $immutableQuote
+     * 
+     * @throws LocalizedException
+     * @throws WebApiException
+     */
+    protected function setShipment($shipment, $immutableQuote)
+    {
+        $shippingAddress = $immutableQuote->getShippingAddress();
+        $address = $shipment['shipping_address'];
+        $address = $this->cartHelper->handleSpecialAddressCases($address);
+        $region = $this->regionModel->loadByName(ArrayHelper::getValueFromArray($address, 'region', ''), ArrayHelper::getValueFromArray($address, 'country_code', ''));
```

Comment:
Looks like method `ArrayHelper::getValueFromArray` makes much more work that needed in this case. You need just check if an array has a key and return '' otherwise, correct? Could you create a helper function that does especially this thing?

Reply:
@vitaliyreznikov Initially the plan is to do refactoring on DiscountCodeValidation.php and the $address could be either array or object, that'why I use ArrayHelper::getValueFromArray here. Now that we do not refactor the related code in DiscountCodeValidation.php, the $addreess is just array, I will update this line

Reply:
@vitaliyreznikov Although getValueFromArray has some more logic to check object, it only executes a few lines for array parameter before returning value, and if we create another function only for checking array, it seems to be redundant. So lets keep as it is for now.

Reply:
Yes, I realized you didn't introduce this, just moved from another place.
Let's postpone the refactoring then.